### DISCUSSION TITLE
difference between "filter selected nodes" and "jump in"

### DISCUSSION THREAD

**Discussion by kenfhill84083** [freeplane/freeplane#1183] [Help] [answered]
Is there a difference between "filter selected nodes" and "jump in"?  

The only difference I can see is that you can filter on multiple nodes and see their ancestors & descendants - but jump in only shows a single node (like a root) with descendants but no ancestors.

Are there any other distinctions?

Is jump in a kind of subset of filter selected nodes?

**Comment by macmarrum** [comment, marked as answer]
Jump in and filter are two separate functionalities. They can be applied one after another.

From https://github.com/freeplane/freeplane/discussions/560#discussioncomment-3450754
 > ViewRoot is not a filter but a different feature. If you jump in / out, the filter is kept.

From a scripting perspective, `node.isVisible() == true` even when the node is outside of a jump-in area.

**Reply by kenfhill84083** [reply]
> Jump in and filter are two separate functionalities. They can be applied one after another.
> 
Thanks.  It looks like both restrict the scope of Find-Replace which is logical.   

Do you have any best practices you've worked out to best use both together?

**Reply by macmarrum** [reply]
After Jumping In, the visible root node behaves much line a regular root node. It can have its children located at both its sides. This cannot be achieved with a filter.

Additionally, Jump In remembers children's sides, so that Jumping Out and back In restores the "local" layout.

Additionally, filter always displays the true root node. Jump In hides the true root node.

**Reply by kenfhill84083** [reply]
> After Jumping In, the visible root node behaves much line a regular root node. It can have its children located at both its sides. This cannot be achieved with a filter.
> 
> Additionally, Jump In remembers children's sides, so that Jumping Out and back In restores the "local" layout.
> 
> Additionally, filter always displays the true root node. Jump In hides the true root node.

Definitely cool.  Thanks.

**Comment by sifran-github** [comment]
I understand what you mean. When I heard about the new "jump in" feature, I also thought to myself "but technically it can be achieved by  "filter selected nodes" so what's the point?

But I think that jump in provides a clear way of focusing on a specific node. You do not need to configure "show ancestors" and "show descendants" every time, and also you can jump-in and continue to filter from there.
Its just a feature which is easy to use, with a clear functionality.

Overall, because this is a feature that is commonly used (in my opinion), its more practical to have it as a separate feature, in addition to the filtering mechanism.

**Comment by kenfhill84083** [comment]
Thanks for the counsel.  I am preparing a video to publish on youtube that introduces Jump In and compares it with Filter Selected Nodes.  I'd like your suggestions.  Several of my videos got posted only to find out later about some aspect I should have included.

The gist of the script is 
- both : isolate nodes for better focus
- FSD : shows node(s) + ancestors + descendants
- JI : only show node + descendants
- FSD : filter multiple nodes
- JI : place node on both sides of focused "root"  +  remembers it
- both : can be used together , simultaneously

Any remaining functionality I need to know?  Any tricks or use cases?

Thanks.

**Reply by sifran-github** [reply]
I would suggest mentioning that the goals of these features are different.

if, for example, you have a big map with many projects:
- you will use filters if you want to filter specific nodes for all projects.
- but if you want to zoom into a specific project and work on it for a while (changing nodes, filtering nodes for this specific project etc') you will first use the jump in feature, work from there on this project, and then jump out. Technically it might be done by combining filters (filter specific node + another filter) but its more complicated.

**Reply by kenfhill84083** [reply]
> I would suggest mentioning that the goals of these features are different.
> 
Thanks.  That's good stuff.

**Comment by kenfhill84083** [comment]
I just discovered you can have cascading Jump Ins.
- Jump in at level 1
- move to level 3, jump in again
- move to level 5, jump in again
- ...

Then Jump Out just unrolls the Jump Ins until you are back where you started.  Cool, robust.

**Reply by sifran-github** [reply]
yeah its really nice, @dpolivaev made a great implementation

and also I asked for the implementation that you will be able to easily jump in nodes even if they are folded. so again, if you have a list of projects you can have them all folded (to have a clean screen), and then you can jump into a specific project when you want to work on it.... and after jumping in you can work on it, and then jump out. it is smooth and keeps the node folded when you jump out. (ref: https://github.com/freeplane/freeplane/discussions/928).

**Comment by kenfhill84083** [comment]
I've made a mindmap to compare the 2 features in something like a table.  Does it look accurate or does it need fixing.


![Freeplane _ Jump in vs Filter selected nodes - table](https://github.com/freeplane/freeplane/assets/6380301/7b838be8-1f1b-4c68-92bb-b1434ee6132a)

**Reply by kenfhill84083** [reply]
> Does it look accurate or does it need fixing.
> 
fixed "ancestor" to "descendant(s)" @ focused "node(s) = descendant(s) of map root"

**Reply by EdoFro** [reply]
imho, you have to choose: use a table or use a mindmap. but trying to simulate a table
with a mindmap is not intuitive for the reader. 

We have to understand that a beginner is just discovering what mind maps are, and one of the first things to learn is that a parent contains children that are related to it, not definitions of other topics. 
jm2c. hope it helps

**Reply by sifran-github** [reply]
I agree with @EdoFro
Tables, specially for comparison, are different kind of data structure, which is not represented well by a mind map. It is also a simpler structure, and many programs can do it. hacking one structure into another only makes it confusing.

**Comment by kenfhill84083** [comment]
Thanks for the tips @sifran-github and @sifran-github RE table-like maps.  I guess I am not certified on all the best practices.  I've changed it to a 2-sided map.  

Some of questions:
- does the information seem accurate?
- would it look better to right-align the left side to be closer to their right-side counterparts?
- do you think the "as edge" connectors work or do they add confusion?  I put them in because there are nodes at the bottom that are not feature-matching.







![Freeplane _ Jump in vs Filter selected nodes - 2-sides](https://github.com/freeplane/freeplane/assets/6380301/cca63b92-5c91-4d9c-9840-0cae0f1c2763)

**Reply by sifran-github** [reply]
I'm not trying to be negative or anything, but since you ask, I will tell you my opinion:

If i had to compare features and present it in an easy-to-understand way, it will be like this:

![image](https://github.com/freeplane/freeplane/assets/5730355/4d04fc82-887c-423d-b36f-143e64b80753)

or something similar.
just a simple table in excel/powerpoint/whatever.

its hard for my mind to understand the mind map that you posted.
But... maybe we just think in different ways and that's ok. I'm just trying to help.

**Reply by kenfhill84083** [reply]
> I'm not trying to be negative or anything, but since you ask, I will tell you my opinion ...

Thanks.  I did not take it negatively.  I appreciate the feedback.

**Reply by kenfhill84083** [reply]
> ![image](https://user-images.githubusercontent.com/5730355/242292412-4d04fc82-887c-423d-b36f-143e64b80753.png)
> 

Made mine in a 3-column table in RightNote.

![Freeplane _ Jump in vs Filter selected nodes - 3-col table](https://github.com/freeplane/freeplane/assets/6380301/df9673f0-732c-462f-bce6-ca8cb94f0b5d)

**Reply by sifran-github** [reply]
Nice. To me it looks clearer this way.

and in order to keep it clear, I would emphasis the purpose of each feature in the beginning of the explanation:

- Jump in: A complete feature, with the intended purpose of allowing the user to focus and work on a specific node. (jump in->focus and work on content->jump out).

- Filter selected nodes: More like a "temporary focus" on a node/nodes. It is just a small part of the entire sophisticated Freeplane filtering system, but it does allows similar functionality by allowing the user to manually select and filter specific nodes. it was a way to utilize something similar to a "jump in" feature before it was developed.

**Reply by sifran-github** [reply]
by the way, @kenfhill84083, do you recommend trying RightNote? if so, why?

I'm hesitant because:
- I cant find any info about this program. no youtube videos... nothing.
- Its not free and not open source.
- In the past I tried several programs that look similar but I didn't like them.

**Reply by kenfhill84083** [reply]
>do you recommend trying RightNote? if so, why?
> 
**like , why :**

I like it a lot.  I got started in 2-pane organizers with KeyNote but the developer went "intellectually bankrupt" - he later said RightNote was the program he always wanted to write.  RN uses several kinds of notes (ie, richtext, richview, spreadsheet, journal, Evernote-linked, memo, source code) - all in one file.  I mostly use RichView for its table support - best app for tables I've ever used.  Can handle tons of data.  Good hyperlinking.  Good Search (but FP is easier-better).  Customer support is good.  Very keyboard friendly.  Fast.  I hope this developer does not go bankrupt in any sense.  Development is active but nothing like FP (but FP is in a development class by itself).

If there was a seamless integration with FP I would be in info heaven.  It has linking to open RN notes from outside RN - like FP - so there is a kind of integration between them.

I use RightNote for bulk-text stuff and when interfacing with others who are mindmap allergic - RN is a little more familiar looking.  And I can't seem to make FP do block outlining the way I like it.

I'd recommend it as a 2-pane organizer.  But I'd feel bad if you bought it based on my recommendation and were disappointed or it ever stopped development.


**your reservations :**

Valid concerns.  It is a 1-man project.  There is no user forum - so every bug/enhancement request is handled via email without any user input.  I have also tried many other 2-pane outliners before (on trial basis) but always come back to RightNote.  RN has a trial-free version.

**Reply by kenfhill84083** [reply]
> and in order to keep it clear, I would emphasis the purpose of each feature in the beginning of the explanation:
> 
>     * Jump in: A complete feature, with the intended purpose of allowing the user to focus and work on a specific node. (jump in->focus and work on content->jump out).
> 
>     * Filter selected nodes: More like a "temporary focus" on a node/nodes. It is just a small part of the entire sophisticated Freeplane filtering system, but it does allows similar functionality by allowing the user to manually select and filter specific nodes. it was a way to utilize something similar to a "jump in" feature before it was developed.

Thanks again for your valuable feedback and suggestions.

I will have to think about how to incorporate this detailed info.
- in the table  or
- in the video narration

I like to populate tables like I do mindmaps - keywords or short phrases where possible.  If this were ever to be used as stand-alone documentation I'd have to factor that in.  I'll play around with some options.

Again, thanks.

**Comment by kenfhill84083** [comment]
This is the final version of the table (at least the last posting here).  I tried to incorporate suggestions offered here - thanks.  I also added access beyond the menus - filter toolbar and Command Search (my FP best friend).  I still welcome further suggestions but I won't post this again.

![Freeplane _ Jump in vs Filter selected nodes - 3-col table](https://github.com/freeplane/freeplane/assets/6380301/be983409-af26-491f-babb-7a58b40d8660)

**Comment by kenfhill84083** [comment]
FYI . I posted the [video](https://www.youtube.com/watch?v=CJm-19pxQxo).  I hope I did not garble it too badly.  I suspect it will not satisfy everyone.

I really enjoyed the interaction here.  It greatly helped clarify both features for me.   Thanks.

Whatever the final result's quality, it is much better than it would have been without feedback here.  I'd like to try this process again on future videos.  After posting several previous videos I later learned of some key information I hadn't known about that would have made a big difference in the videos' accuracy.

**Comment by kenfhill84083** [comment]
FYI . I posted the [video](https://www.youtube.com/watch?v=CJm-19pxQxo).  I hope I did not garble it too badly.  I suspect it will not satisfy everyone.

I really enjoyed the interaction here.  It greatly helped clarify both features for me.   Thanks.

Whatever the final result's quality, it is much better than it would have been without feedback here.  I'd like to try this process again on future videos.  After posting several previous videos I later learned of some key information I hadn't known about that would have made a big difference in the videos' accuracy.

**Reply by sifran-github** [reply]
I watched it. nice and clear.

I created this feature request to add jump in and jump out to the toolbar (https://github.com/freeplane/freeplane/issues/1192) which i wanted to create a long time ago but your video reminded me.

**Reply by kenfhill84083** [reply]
> FYI . I posted the [video](https://www.youtube.com/watch?v=CJm-19pxQxo). 

One hopefully final note.  I posted a [supplemental video](https://studio.youtube.com/video/ilRdcSHQBI8/edit).
- you can access Jump In/Out from the right-click menu
- you can "back-out" of Filter Select Nodes by going node-by-node in the ancestors re-invoking the filter.

I uploaded a revised table to [dropbox](https://www.dropbox.com/s/rh7kwq3sdm5683a/Freeplane%20_%20Jump%20in%20vs%20Filter%20selected%20nodes%20-%203-col%20table.jpg?dl=0).

**Reply by sifran-github** [reply]
this is one of my favorite parts about jump in:
https://github.com/freeplane/freeplane/discussions/1183#discussioncomment-6039601

The ability to jump into a folded node, edit its children and jumping out. works smoothly.
You cant do that with filters.

**Comment by kenfhill84083** [comment]
>
> retaining fold status
>

Thanks.  Good stuff.  Anything else come to mind?

I updated the table on dropbox.  I will make another supplement video - I
will wait a bit to see if any new gems surface.

**Reply by sifran-github** [reply]
> Anything else come to mind?

That's it from my end. just this ability to keep some projects nodes folded (for example) and jumping into one of them when wanting to work on it.

**Reply by sifran-github** [reply]
@kenfhill84083 i just posted an example of what I mean, here: https://github.com/freeplane/freeplane/discussions/1197. I wanted to write this quick tip about it anyway.
